### PR TITLE
Enhancement/Update Composer Dependencies

### LINKED ISSUE #507: Security vulnerability detected
See

* https://www.versioneye.com/user/projects/54885d5a746eb514b0000279#tab-security

Maybe someone can take a look?

### PR DESCRIPTION
**Pull request by ins0**
This PR

- [x] jumps to the latest stable dependencies versions
 - ZF2 v2.5
 - scn-social-auth v1.18.1
 - htmlpurifier v4.7.0
 - zfc-user v1.3.3
- [x] fixes followed errors due to major dependencies upgrades
- [x] adds ``UserController`` test case
- [x] replace the composer meta package ``zendframework/zendframework`` with the separated zend packages (generally replace - nothing sorted out)
- [x] increased required php version to ``5.6`
- [x] adds the ``Roave SecurityAdvisories`` package
- [x] update the ``composer.lock`` file

fixes #507 

/cc @Ocramius @localheinz

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -4,18 +4,46 @@
...
+        "zendframework/zend-feed": "~2.5.0",
+        "zendframework/zend-math": "~2.5.0",
+        "zendframework/zend-crypt": "~2.5.0",
+        "zendframework/zend-authentication": "~2.5.0",
+        "zendframework/zend-paginator": "~2.5.0",
+        "zendframework/zend-file": "~2.5.0",
+        "zendframework/zend-uri": "~2.5.0",
+        "zendframework/zend-soap": "~2.5.0",
+        "zendframework/zend-db": "~2.5.0",
+        "evandotpro/edp-github": "dev-enhancement/dependencies as 0.3.0",
```
Comment (ins0) [inline composer.json]: this is a workaround as long as https://github.com/EvanDotPro/EdpGithub/pull/42 isn't merged, as composer wouldn't update our dependencies without. 

should stay or should we wait?

File: composer.json [hunk 2]
```diff
@@ -4,18 +4,47 @@
...
     "require": {
-        "php": "~5.5",
+        "php": "~5.6",
         "ext-intl": "*",
         "doctrine/doctrine-module": "^0.9.0",
         "doctrine/doctrine-orm-module": "^0.9.1",
         "doctrine/migrations": "^1.1",
-        "evandotpro/edp-github": "0.2.*",
-        "ezyang/htmlpurifier": "4.6.*",
+        "evandotpro/edp-github": "dev-enhancement/dependencies as 0.3.0",
```
Comment (ins0) [inline composer.json]: this is a workaround as long as https://github.com/EvanDotPro/EdpGithub/pull/42 isn't merged, as composer wouldn't update our dependencies without. 

should stay or should we wait?

**Comment by ins0** [comment]
no clue why scrutinizerhates me :sob:

**Comment by ins0** [comment]
@localheinz or @Ocramius could you please have a look :smiley:

File: composer.json (hunk 2 above)
Comment (Ocramius) [inline composer.json]: @ins0 why doesn't a stable version work? I don't see any particular differences

File: module/User/src/User/Controller/UserController.php [hunk 3]
```diff
@@ -19,10 +19,15 @@ class UserController extends ZfcUserController
     private $moduleService;
 
     /**
+     * @param callable $redirectCallback
      * @param Service\Module $moduleService
      */
-    public function __construct(Service\Module $moduleService)
+    public function __construct($redirectCallback, Service\Module $moduleService)
     {
+        if (!is_callable($redirectCallback)) {
```
Comment (Ocramius) [inline module/User/src/User/Controller/UserController.php]: This needs a test case

**Comment by gianarb** [comment]
@ins0 edpgithub 0.4.0 is ready :)

File: composer.json (hunk 2 above)
Comment (ins0) [inline composer.json]: because the stable version required zend 2.2 but the pr is merged so i can update this

**Comment by ins0** [comment]
thanks for the info @gianarb i will update this :+1:

**Comment by Ocramius** [comment]
@ins0 any changes needed? I assume we don't need to alias the requirements anymore..

**Comment by ins0** [comment]
@Ocramius unfortunately the packagist repo isn't updated, so @gianarb could you do this too or did we need to ask @EvanDotPro to do this please?

**Comment by Ocramius** [comment]
It's up to @Hounddog

**Comment by ins0** [comment]
thanks @Hounddog 

ping @Ocramius 
don't know why travis not showing a green build, but its run green like the latest runs too. :8ball:

**Comment by Ocramius** [comment]
@ins0 had this issue with travis also with other builds, I think it's not on our side atm.

**Comment by Ocramius** [comment]
Thanks for your effort on this, @ins0!

**Comment by ins0** [comment]
thanks @Ocramius

**Pull request merged**